frontend/src/metabase/lib/redux/typed-utils.ts

```diff
@@ -11,13 +12,26 @@ interface ThunkConfig {
 export const createAsyncThunk =
   createAsyncThunkOriginal.withTypes<ThunkConfig>();
 
+type AwaitedIfPromise<T> = T extends Promise<infer R> ? R : T;
```

Comment:
This is needed because even if createThunkAction awaits the call to the payload creator, most of the times we don't actually pass an async function to it

Reply:
Could we use TypeScript's built-in `Awaited` type instead? It's built-in as of [TypeScript 4.5](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#the-awaited-type-and-promise-improvements). I think it has the same definition as yours, just with `PromiseLike` instead of `Promise`

Reply:
Ah didn't know it acted the same, I knew ts had something like that but I thought it only worked with Promise<T> not with T | Promise<T>

Thanks!